lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -687,6 +688,116 @@ def process(self, calibrated_forecast_predictor_and_variance,
+               'only a leading threshold dimension, followed by spatial (y/x) '
+               'dimensions. Got dimensions: {}'.format(dim_coords))
+        raise ValueError(msg)
+
+    def _mean_and_variance_to_probabilities(self, mean_values, variance_values,
+                                            probability_cube_template):
+        """
+        Function returning probabilities relative to provided thresholds based
+        on the supplied mean and variance. A Gaussian distribution is assumed.
+
+        Args:
+            mean_values (iris.cube.Cube):
+                Predictor for the calibrated forecast i.e. the mean.
+            variance_values (iris.cube.Cube):
+                Variance for the calibrated forecast.
+            probability_cube_template (iris.cube.Cube):
+                A probability cube that has the threshold coordinate, and
+                attribute relative_to_threshold, that match the desired output
+                cube format.
+
+        Returns:
+            probability_cube (Iris cube):
+                Cube containing the data expressed as probabilities relative to
+                the provided thresholds in the way described by
+                relative_to_threshold.
+        """
+        thresholds = probability_cube_template.coord('threshold').points
+        relative_to_threshold = (
+            probability_cube_template.attributes['relative_to_threshold'])
+
+        # Loop over thresholds, and use a normal distribution with the mean
+        # and variance to calculate the probabilties relative to each
+        # probability.
+        probabilities = np.empty_like(probability_cube_template.data)
+        distribution = norm(loc=mean_values.data,
+                            scale=np.sqrt(variance_values.data))
+        probability_method = distribution.cdf
+        if relative_to_threshold == 'above':
+            probability_method = distribution.sf
+
+        for index, threshold in enumerate(thresholds):
+            probabilities[index, ...] = probability_method(threshold)
+
+        probability_cube = probability_cube_template.copy(data=probabilities)
+        return probability_cube
+
+    def process(self, mean_values, variance_values, probability_cube_template):
+        """
+        Generate ensemble percentiles from the mean and variance.
+
+        Args:
+            mean_values (iris.cube.Cube):
+                Cube containing the distribution mean values of a diagnostic,
+                e.g. the mean over realizations.
+            variance_values (iris.cube.Cube):
+                Cube containing the distribution variance values of a
+                diagnostic, e.g. the variance across realizations.
+            probability_cube_template (iris.cube.Cube):
+                A probability cube that has the threshold coordinate, and
+                attribute relative_to_threshold, that match the desired output
+                cube format.
```

Comment:
Should we check the units somewhere?  I can see the envisaged usage, but it would be good to make sure that if eg the probability cube thresholds were in degrees C and the mean and variance in Kelvin that we could handle that (or throw an error if need be).

Reply:
I think this is a good shout. I'll make some changes now.

Reply:
Done.